sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -199,7 +199,12 @@ case class Divide(left: Expression, right: Expression) extends BinaryArithmetic
 
   override def symbol: String = "/"
   override def decimalMethod: String = "$div"
-  override def nullable: Boolean = true
+  override def nullable: Boolean = (left.nullable ||
+    !(!right.nullable && right.isInstanceOf[Literal] &&
+      ((right.asInstanceOf[Literal].value.isInstanceOf[Decimal] &&
```

Comment:
This is too complicated, do you think it worth to save the the nullable-checking?

If it does not have good speed up (by benchmark), I'd like to not do this.